src/box/memtx_tx.c

```diff
@@ -2024,20 +2030,19 @@ memtx_tx_on_space_delete(struct space *space)
 			= rlist_first_entry(&space->memtx_stories,
 					    struct memtx_story,
 					    in_space_stories);
+		/*
+		 * Space is to be altered (not necessarily dropped). Since
+		 * this operation is considered to be DDL, all other
+		 * transactions will be aborted anyway. We can't postpone
+		 * rollback till actual call of commit/rollback since stories
+		 * should be destroyed immediately.
+		 */
+		if (story->add_stmt != NULL)
+			memtx_tx_history_rollback_stmt(story->add_stmt);
```

Comment:
Could you please examine that `memtx_tx_history_rollback_stmt` does exactly what you want? 
For instance shoud `memtx_tx_story_full_unlink` be called after that or not necessary?

Reply:
It seems to be OK: while we are rolling back mvcc stmt GC anyway calls `full_unlink + story_delete`. So as in this case we also rollback stmt, these pair of cleanup functions look suitable.